src/app/navbar/navbar.component.ts

```diff
@@ -12,9 +12,18 @@ export class NavbarComponent implements OnInit {
     @Output() changeView: EventEmitter<string> = new EventEmitter();
 
     constructor(private _eventService: EventService, private _categService: CategoryService) { }
-    ngOnInit() { }
-
+    temperature: string;
+    icon: string;
     isCollapsed: boolean = true;
+    ngOnInit() { 
+      // our call back function 
+      var tempCallback = function callback(temperature, icon) : void{
```

Comment:
Not sure we need to wrap this in a callback, can just call a function that sets the variables themselves (e.g. just make this.getTemperature() directly modify this.temperature and this.icon